src/main/java/org/zalando/riptide/Navigator.java

```diff
@@ -22,26 +22,25 @@
 
 import org.springframework.http.client.ClientHttpResponse;
 
-import javax.annotation.Nullable;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
-/**
- * A {@link Selector} can be used change the dispatching strategy. Its purpose is to find a binding for a given
- * response.
- *
- * @param <A> generic binding attribute type
- */
-public interface Selector<A> {
-
-    /**
-     * Attempts to find a matching binding for the given response.
-     *
-     * @param response the received response
-     * @param routes  a map containing all routes
-     * @return an optional binding match, if found
-     */
-    @Nullable
-    Executor select(final ClientHttpResponse response, final Map<A, Executor> routes) throws IOException;
+@FunctionalInterface
+public interface Navigator<A> {
+
+    Optional<Route> navigate(final ClientHttpResponse response, final RoutingTree<A> tree) throws IOException;
+
+    default List<Binding<A>> merge(final List<Binding<A>> present, final List<Binding<A>> additional) {
```

Comment:
It's part of the interface on purpose. Merging bindings/trees is potentially very dependent on the semantics of the navigator. First-write wins, last-write wins, overwrite, ...